### ISSUE TITLE
Wrong repair cost for Ranger program

### ISSUE THREAD

**Issue by rnyoakum** [raceintospace/raceintospace#204]
I encountered a news event (The Space Program has made a design error on the Ranger satellite program...) specifying that repairs would have to be made for 10MB or face a 10% penalty on the next flight. However, the repair option popup after the news event was for a 20MB / 20% penalty, and charged accordingly. In the audio dialogue, Carter Walcrite specified it was a 10% penalty.

**Comment by rnyoakum** [comment]
I looked into this further, and it appears the news event is implemented correctly and the program is damaged properly. The problem here appears to be with the autosave.

The autosave is created at the end of the turn, recording the program state at that time. This includes the applied effect of the news event. But when the autosave is loaded and the start-of-turn news broadcast is shown, the event is reapplied. In this case, the -10%/10MB penalty is added cumulatively to the previously applied damage/cost variables, hence the -20%/20MB penalty. If the player had paid to fix the issue, this bug would have reintroduced the -10%/10MB damage - but not refunded the MBs already paid. As it stands now, this is an issue that will affect a wide variety of news events, both negative (program damage) to positive (bonus cash).

The same autosave news-reprocessing also applies to recording news events, creating duplicate news reports which are visible in the viewing stand. Each time the news broadcast is shown, it gets recorded for posterity. The game only expects this to happen once per turn, so when an autosave is loaded and the news is displayed again the second occurrence is also recorded.

As for the hardware damage code, I looked it over and it should work as intended; once this problem is fixed (I'm currently working on it) we'll have to be alert in case there are other bugs messing with the repair costs.
